yt_dlp/extractor/fox.py

```diff
@@ -87,16 +89,37 @@ def _real_initialize(self):
             if not self._access_token:
                 self._access_token = self._call_api(
                     'login', None, json.dumps({
-                        'deviceId': compat_str(uuid.uuid4()),
+                        'deviceId': self._device_id,
                     }).encode())['accessToken']
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
-        video = self._call_api('vodplayer/' + video_id, video_id)
+        self._access_token = self._call_api(
+            'previewpassmvpd?device_id=%s&mvpd_id=TempPass_fbcfox_60min' % self._device_id,
+            video_id)['accessToken']
+
+        video = self._call_api('watch', video_id, data=json.dumps({
+            'capabilities': ['drm/widevine', 'fsdk/yo'],
```

Comment:
why are we asking for DRM formats?

Reply:
That's a good question that even I don't know the answer to. All I can say is that that was the post request sent to the API, and that none of the formats in the m3u8 that is eventually returned are DRMed. Perhaps the nature of this request is a relic from when Fox did use DRM or it is in preparation for some future date when they do implement DRM. If you like, I try to snoop the requests in a non-DRM browser (e.g. Firefox) and see if the request changes. But I also wouldn't put too much stock into the data sent here anyway, for example max resolution is listed as 720p, but if a video has 1080p formats available, they will still be returned. It seems the only things that really make a difference (aside from the structure of the data) are the device ID, which must match the ID used in the "login" request at initialization, and the stream ID, which must match the video ID